spring-boot-starter-mesos/src/main/java/com/containersolutions/mesos/scheduler/FrameworkInfoFactory.java

```diff
@@ -0,0 +1,44 @@
+package com.containersolutions.mesos.scheduler;
+
+import com.containersolutions.mesos.scheduler.config.MesosConfigProperties;
+import com.containersolutions.mesos.scheduler.state.StateRepository;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.mesos.Protos;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+
+/**
+ * Creates framework info
+ */
+public class FrameworkInfoFactory {
+    protected final Log logger = LogFactory.getLog(getClass());
+
+    @Value("${spring.application.name}")
+    protected String applicationName;
+
+    @Autowired
+    MesosConfigProperties mesosConfig;
+
+    @Autowired
+    StateRepository stateRepository;
+
+    @Autowired
+    CredentialFactory credentialFactory;
+
+    public Protos.FrameworkInfo.Builder create() {
+        Protos.FrameworkInfo.Builder frameworkBuilder = Protos.FrameworkInfo.newBuilder()
+                .setName(applicationName)
+                .setUser("root")
```

Comment:
@justinholmes, this will conflict with your change, https://github.com/justinholmes/mesos-starter/commit/1b39169a89e19c38564847edd002130f61a966dc#diff-a8ae72aff7e495716601eb3da9734e94R71